src/Compress/Bz2.php

```diff
@@ -4,183 +4,66 @@
 
 namespace Laminas\Filter\Compress;
 
-use Laminas\Filter\Exception;
+use Laminas\Filter\Exception\ExtensionNotLoadedException;
+use Laminas\Filter\Exception\RuntimeException;
 
-use function bzclose;
+use function assert;
 use function bzcompress;
 use function bzdecompress;
-use function bzopen;
-use function bzread;
-use function bzwrite;
 use function extension_loaded;
-use function file_exists;
 use function is_int;
-use function str_contains;
+use function sprintf;
 
 /**
  * Compression adapter for Bz2
  *
  * @psalm-type Options = array{
- *     blocksize?: int,
- *     archive?: string|null,
+ *     blocksize?: int<1, 9>|null,
  * }
- * @extends AbstractCompressionAlgorithm<Options>
  */
-final class Bz2 extends AbstractCompressionAlgorithm
+final class Bz2 implements StringCompressionAdapterInterface
```

Comment:
@gsteel 
They are not filters, but adapters for filters. Unfortunately, this is not clear from the name of the class or namespace. Only the interface indicates this.
I think it would be helpful to make this clearer somehow.

Reply:
Considering they are complete rewrites too, I agree that renaming them is probably a good idea 👍